doc/ioc/tutorials/streamdevice.md

```diff
@@ -0,0 +1,288 @@
+It also specifies how they relate to the functions written in the protocol file.
+
+``` {.perl filename="exampleApp/Db/example.db"}
+record(ai, "${PREFIX}VOLT-RB") {
+    field(DTYP, "stream")
+    field(INP, "@example.proto getVoltage ${PORT}")
+}
+
+record(ao, "${PREFIX}VOLT") {
+    field(DTYP, "stream")
+    field(OUT, "@example.proto setVoltage ${PORT}")
+}
+```
+
+Change `exampleApp/Db/Makefile`
+so that the EPICS build system installs `example.proto` and `example.db`:
+
+``` {.makefile filename="exampleApp/Db/Makefile"}
+# ...
+
+#----------------------------------------------------
+# Create and install (or just install) into <top>/db
+# databases, templates, substitutions like this
+DB += example.db
+DB += example.proto
+
+# ...
+```
+
+Change your `st.cmd` file
+so that it knows where to load the protocol file,
+and how to connect to the remote power supply.
+
+``` {.csh filename="iocBoot/iocExample/st.cmd"}
+#!../../bin/linux-x86_64/example
+
+< envPaths
+
+## Register all support components
+dbLoadDatabase("${TOP}/dbd/example.dbd")
+example_registerRecordDeviceDriver(pdbbase)
+
+# Where to find the protocol files
+epicsEnvSet("STREAM_PROTOCOL_PATH", "${TOP}/db")
+# The TCP/IP address of the power supply
+drvAsynIPPortConfigure("PS1", "localhost:8727")
+
+## Load record instances
+dbLoadRecords("${TOP}/db/example.db", "PREFIX=, PORT=PS1")
+
+iocInit()
+```
+
+And run `chmod +x iocBoot/iocExample/st.cmd`
+so that you can run your command file as-is.
+
+You can test that your top builds by running:
+
+``` bash
+nix build -L
+```
+
+You will see that your IOC does not build.
+This is because we haven't told Git to track those newly added files,
+and so Nix ignores them too.
+
+Run `git add .`{.sh} for Git and Nix to track all files,
+and try a `nix build -L`{.sh} again.
+
+If everything goes right,
+you can examine your compiled top under `./result`.
+
+You can observe that:
+
+-   the `example` app is installed under `bin/` and `bin/linux-x86_64`,
+    and links to the correct libraries
+-   `example.proto` and `example.db` are installed under `db/`
+-   `example.dbd` is generated and installed under `dbd/`
+
+# Running your IOC
+
+To run your IOC,
+build it first with `nix build -L`{.sh},
+and place yourself in the `./result/iocBoot/iocExample` directory.
+Then, run:
+
+``` bash
+./st.cmd
+```
+
+You should see the IOC starting and connecting to `localhost:8727`.
+
+# Recompiling with make
+
+Using `nix build`{.sh} to compile your IOC each time might feel slow.
+This is because Nix re-compiles your IOC from scratch each time.
+
+If you want a more "traditional" edit / compile / run workflow,
+you can place yourself in the development shell with `nix develop`{.sh},
+and use `make` from here.
+
+# Next steps
+
+More commands are available in the power supply simulator.
+To view them,
+close your IOC,
+and open a direct connection to the simulator:
+
+``` bash
+nc localhost 8727
+```
+
+In this `nc`{.sh} prompt,
```

Comment:
nc is more standard than telnet ?

Reply:
netcat is installed by default in my distribution, and not telnet. Not sure about other Linux distributions.

Reply:
I've found that at least in CentOS 7 and Rocky 9, neither netcat nor telnet are installed by default. I'll add a line to specify which package to install.